src/inmanta/data/model.py

```diff
@@ -722,3 +722,19 @@ class EnvironmentMetricsResult(BaseModel):
     end: datetime.datetime
     timestamps: List[datetime.datetime]
     metrics: Dict[str, List[Optional[Union[float, Dict[str, float]]]]]
+
+
+class User(BaseModel):
+    """A user"""
+
+    username: str
+    auth_method: str
```

Comment:
I agree with you. This is not free-form so I don't think we should model it as such either. I would actually even extend that to the database. If we want to avoid enums there I would go for the approach Markus suggested for this, iirc he said to create a separate table that contains all valid auth_methods together with a numeric id, and to use that id in this table. Then again, I can't say off the top of my head how well this would fit in our current data model.

Reply:
Yes. Indeed. I am not sure either whether this will fit our current model well. Maybe we have to go for enums in the database for now.

Reply:
I'm inclined to agree.